src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```
@@ -2,13 +2,29 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
```

You check “duplicate login” before “login length.” That’s OK, but if someone passes a 3‑char login that matches an existing user’s short login, you’ll throw “User already exists” rather than “Login too short.” Consider checking null/length first, then uniqueness, so users always see the most relevant error